iterreg/sparse/solvers.py

```diff
@@ -167,15 +167,18 @@ def cd_primal_dual(X, y, prox=shrink, max_iter=100, f_store=1, verbose=False):
 
 @njit
 def cd(X, y, alpha, prox=shrink, pen=ell1, max_iter=1_000,
-       f_store=1, verbose=False):
+       f_store=1, verbose=False, w_0=None):
     """Coordinate descent for the Tikhonov problem."""
     p = X.shape[1]
     X = np.asfortranarray(X)
     lc = np.zeros(p)
     for j in range(p):
         lc[j] = norm(X[:, j]) ** 2
     R = y.copy().astype(np.float64)
-    w = np.zeros(p)
+    if w_0 is None:
```

Comment:
changing w's type from None to a np.array does not break numba ?

Reply:
It seems to work, I had no error message